docs/typescript/examples/task-scheduler.md

```diff
@@ -0,0 +1,574 @@
+- [Knex](https://knexjs.org/) – Type-safe database access and schema management
+- [DBOS Scheduled Workflows](../tutorials/scheduled-workflows.md) – Ensure tasks are run as scheduled
+- React, with [Material](https://mui.com) and [react-big-calendar](https://github.com/jquense/react-big-calendar) – Present a calendar of tasks and results
+- Next.js server actions – Simple interaction between the browser-based client and the server
+- Next.js API routes and DBOS HTTP endpoints – Allow access to the server logic from clients other than Next.js
+- [WebSockets](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API) – Send calendar and result updates to the browser with low latency
+- Database triggers – Listen for database updates made by other VMs
+- [Jest](https://jestjs.io/) – Unit test backend code
+
+## DBOS and Database Logic
+
+### Task Code
+The list of schedulable tasks is in `src/dbos/tasks.ts`. The `schedulableTasks` array contains the available tasks, with information needed for `doTaskFetch` to execute them.  Tasks can be added by expanding the array with additional entries:
+```typescript
+  {
+    id: 'fetch_joke', // Unique ID for the task
+    name: 'Fetch Random Joke', // Text label the task
+    url: 'https://official-joke-api.appspot.com/random_joke', // URL to fetch when the task runs
+    type: 'json', // Type of result to expect from the task
+  },
+```
+
+### Main Workflow
+The main workflow for executing tasks is in `src/dbos/operations.ts`, in the `SchedulerOps` class:
+
+```typescript
+  @DBOS.workflow()
+  static async runJob(sched: string, task: string, time: Date) {
+    DBOS.logger.info(`Running ${task} at ${time.toString()}`);
+
+    let resstr = "";
+    let errstr = "";
+
+    try {
+      // Fetch the result
+      const res = await SchedulerOps.runTask(task);
+      resstr = res;
+
+      // Store result in database
+      await ScheduleDBOps.setResult(sched, task, time, res, '');
+    }
+    catch (e) {
+      const err = e as Error;
+      // Store error in database
+      await ScheduleDBOps.setResult(sched, task, time, '', err.message);
+      errstr = err.message;
+    }
+
+    // Tell attached clients
+    SchedulerOps.notifyListeners('result');
+
+    // Send notification
+    await SchedulerOps.sendStatusEmail(
+      errstr ? `Task ${task} failed` : `Task ${task} result`,
+      errstr || resstr
+    );
+  }
+```
+
+While it may be overkill for this demo application, `runJob` will be executed durably.  That is, if the server crashes after `runTask` is complete, but the result hasn't been recorded in the database with `setResult`, or if the email hasn't been sent by `sendStatusEmail`, DBOS Transact will finish the workflow during recovery and execute those steps.
```

Comment:
```suggestion
Because `runJob` is a DBOS workflow, it is executed durably.  That is, if the server crashes after `runTask` is complete, but the result hasn't been recorded in the database with `setResult`, or if the email hasn't been sent by `sendStatusEmail`, DBOS Transact will finish the workflow during recovery and execute those steps.
```

Being more positive :)